src/presentational-components/shared/table-toolbar-view.js

```diff
@@ -1,150 +1,127 @@
-import { Pagination } from '@redhat-cloud-services/frontend-components';
-import { scrollToTop, getCurrentPage, getNewPage } from '../../helpers/shared/helpers';
+import { getCurrentPage } from '../../helpers/shared/helpers';
 import { defaultSettings  } from '../../helpers/shared/pagination';
-import FilterToolbar from '../../presentational-components/shared/filter-toolbar-item';
-import { TableToolbar } from '@redhat-cloud-services/frontend-components/components/TableToolbar';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/PrimaryToolbar';
 import { ListLoader } from './loader-placeholders';
 import './table-toolbar-view.scss';
 
 export const TableToolbarView = ({
-  request,
-  isSelectable,
   isCompact,
   createRows,
   borders,
   columns,
   toolbarButtons,
-  fetchData,
   data,
   actionResolver,
   areActionsDisabled,
   routes,
   titlePlural,
   titleSingular,
   pagination,
-  checkedRows,
-  setCheckedItems,
   filterValue,
   isLoading,
   setFilterValue }) => {
-  const [ rows, setRows ] = useState([]);
+  const [ opened, openRow ] = useState({});
 
-  useEffect(() => {
-    fetchData(setRows, filterValue, pagination);
-  }, [ filterValue, pagination.limit, pagination.offset ]);
-
-  useEffect(() => {
-    setRows(createRows(data, checkedRows, filterValue));
-    if (isSelectable) {
-      setCheckedItems(rows.filter(item => (item.uuid && item.selected)));
-    }
-  }, [ data ]);
-
-  useEffect(() => {
-    scrollToTop();
-  }, []);
-
-  const handleOnPerPageSelect = limit => request({
-    offset: pagination.offset,
-    limit
-  }).then(({ value: { data }}) => setRows(createRows(data, checkedRows, filterValue)));
-
-  const handleSetPage = (number, debounce) => {
-    const options = {
-      offset: getNewPage(number, pagination.limit),
-      limit: pagination.limit
-    };
-    const requestFunc = () => request(options);
-    return debounce ? debouncePromise(request, 250)() : requestFunc()
-    .then(({ value: { data }}) => setRows(createRows(data, checkedRows, filterValue)));
-  };
-
-  const setOpen = (data, uuid) => data.map(row => row.uuid === uuid ?
-    {
-      ...row,
-      isOpen: !row.isOpen
-    } : {
-      ...row
-    });
-
-  const setSelected = (data, uuid) => {
-    const newData = data.map(row => row.uuid === uuid ?
-      {
-        ...row,
-        selected: !row.selected
-      } : {
-        ...row
-      });
-
-    const checkedItems = newData.filter(item => (item.uuid && item.selected));
-    setCheckedItems(checkedItems);
-    return newData;
-  };
-
-  const onCollapse = (_event, _index, _isOpen, { uuid }) => setRows((rows) => setOpen(rows, uuid));
-
-  const selectRow = (_event, selected, index, { uuid } = {}) => index === -1
-    ? setRows(rows.map(row => ({ ...row, selected })))
-    : setRows((rows) => setSelected(rows, uuid));
+  const onCollapse = (_event, _index, isOpen, { uuid }) => openRow({
+    ...opened,
```

Comment:
I would prefer using `useState` with callback to ensure that `opened` has always valid value
```jsx
openRow(opened => ({...openned, [uuid]: isOpen}))
```